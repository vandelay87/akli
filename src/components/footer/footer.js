import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import LinkCTA from '../link/linkCTA';
import { color } from '../../styles/colors';

const Footer = ({ navigation }) => (
  <StyledFooter>
    <StyledNavigation>
      {navigation.map((link) => (
        <li key={link.id}>
          <LinkCTA title={link.title} link={link.link} external={link.external} />
        </li>
      ))}
    </StyledNavigation>
  </StyledFooter>
);

const StyledFooter = styled.footer`
  background: ${color.secondary};
  padding: 16px;
`;
const StyledNavigation = styled.ul`
  list-style: none;
  padding: 0;
  margin-left: 16px;

  a {
    color: ${color.onSecondary};
  }
`;

Footer.propTypes = {
  navigation: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
    external: PropTypes.bool,
  })).isRequired,
};

export default Footer;
