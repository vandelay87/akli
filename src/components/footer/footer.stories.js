import React from 'react';
import { object } from '@storybook/addon-knobs';
import Footer from './footer';

export default {
  component: Footer,
  title: 'Footer',
};

export const footer = () => {
  const defaultValue = [
    {
      id: '1',
      title: 'Home',
      link: '/',
    },
    {
      id: '2',
      title: 'Google',
      link: 'https://www.google.com',
      external: true,
    },
  ];
  const knob = object('Navigation', defaultValue);

  return <Footer navigation={knob} />;
};
