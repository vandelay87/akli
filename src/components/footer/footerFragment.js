import { graphql } from 'gatsby'

export const query = graphql`
  fragment Footer on ContentfulWebsiteSettings {
    footer {
      navigation {
        ...Link
      }
    }
  }
`
