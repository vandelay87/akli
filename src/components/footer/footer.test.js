import renderer from 'react-test-renderer';
import { footer } from './footer.stories';

describe('footer.js', () => {
  it('renders component', () => {
    const tree = renderer
      .create(footer())
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
