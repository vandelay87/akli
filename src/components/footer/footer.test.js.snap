// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer.js renders component 1`] = `
<footer
  className="sc-bwzfXH iuGtdQ"
>
  <ul
    className="sc-htpNat kjVJHN"
  >
    <li>
      <a
        aria-current="page"
        className="sc-bdVaJa inBooY"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className="sc-bdVaJa inBooY"
        href="https://www.google.com"
        rel="noreferrer noopener"
        target="_blank"
      >
        Google
      </a>
    </li>
  </ul>
</footer>
`;
